pypeit/core/mosaic.py

```diff
@@ -0,0 +1,312 @@
+        overlap (:obj:`str`, optional):
+            Keyword that indicates how to handle pixels in the regions where
+            multiple images overlap in the mosaic.  Options are:
+
+                - ``'combine'``: Average the values of the pixels and, if the
+                  inverse variance is provided, propagate the error.
+                - ``'error'``: Raise an exception.  Largely provided for testing
+                  under the expectation that *no* pixels should overlap in the
+                  mosaic.
+
+    Returns:
+        :obj:`tuple`: Four objects are returned. The first three are
+        `numpy.ndarray`_ objects with the mosaic image, its inverse variance
+        (None if no inverse variance is provided), and an integer array with the
+        number of input pixels in each output pixel.  The last contains the
+        detailed transformation matrices applied to each image.  If
+        ``mosaic_shape`` is provided, these are identical to the input
+        ``tforms``; otherwise, these are the transforms adjusted from the
+        relative frame to the absolute mosaic frame given its determined shape;
+        see :func:`~pypeit.core.mosaic.prepare_mosaic`.
+    """
+    # Check the input
+    nimg = len(imgs)
+    if len(tforms) != nimg:
+        msgs.error('Number of image transformations does not match number of images to mosaic.')
+    if ivar is not None and len(ivar) != nimg:
+        msgs.error('If providing any, must provide inverse-variance for each image in the mosaic.')
+    if bpm is not None and len(bpm) != nimg:
+        msgs.error('If providing any, must provide bad-pixel masks for each image in the mosaic.')
+    if overlap not in ['combine', 'error']:
+        msgs.error(f'Unknown value for overlap ({overlap}), must be "combine" or "error".')
+
+    # Get the output shape, if necessary
+    if mosaic_shape is None:
+        shape = imgs[0].shape
+        if not np.all([img.shape == shape for img in imgs]):
+            msgs.error('If output mosaic shape is not provided, all input images must have the '
+                       'same shape!')
+        mosaic_shape, _tforms = prepare_mosaic(shape, tforms)
+    else:
+        _tforms = tforms
+
+    msgs.info(f'Constructing image mosaic with {nimg} images and output shape {mosaic_shape}.')
+
+    if ivar is not None:
+        var = [inverse(_ivar) for _ivar in ivar]
+
+    mosaic_npix = np.zeros(mosaic_shape, dtype=int)
+    mosaic_data = np.zeros(mosaic_shape, dtype=float)
+    # NOTE: "mosaic_ivar" is actually the variance until it's inverted just
+    # before output
+    mosaic_ivar = None if ivar is None else np.zeros(mosaic_shape, dtype=float)
+
+    # TODO: These loops can end up creating and destroying lots of big arrays.
+    # Is there a way to make this faster?
+    for i in range(nimg):
+        _inv_tform = np.linalg.inv(_tforms[i])
+        # TODO: I'm not a fan of using cval=np.nan, but this might be the most
```

Comment:
I would start with np.nan and if it is annoying, we can discuss a better approach. In these situations I often use a fill value like -1e20 or something. Each has its ups and downsides. I think the main thing you want is to be able to easily identify any bugs where you accidentally operation these values.

Reply:
Changed to -1e20.